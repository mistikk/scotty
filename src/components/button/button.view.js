import React from 'react';
import { Text, TouchableOpacity } from 'react-native';

import styles from './button.styles';

const MakerPopoverView = ({ text, style, onPress }) => (
  <TouchableOpacity style={[styles.container, style]} onPress={onPress}>
    <Text style={styles.text}>{text}</Text>
  </TouchableOpacity>
);

export default MakerPopoverView;
