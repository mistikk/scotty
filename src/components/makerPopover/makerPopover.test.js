import React from 'react';
import renderer from 'react-test-renderer';

import { Button } from '../button';

import MakerPopover from './makerPopover.view';

describe('Maker Popover Component > iOS', () => {
  beforeEach(() => {
    jest.mock('Platform', () => {
      const Platform = require.requireActual('Platform');
      Platform.OS = 'ios';
      return Platform;
    });
  });
  it('> render the component', () => {
    const component = renderer.create(
      <MakerPopover id="1" name="test" distance={4} rating={3} />,
    );

    expect(component).toMatchSnapshot();
  });

  it('> button onClick function', () => {
    const id = '1';
    const mockCallback = jest.fn();
    const component = renderer.create(
      <MakerPopover
        id={id}
        name="test"
        distance={4}
        rating={3}
        handleOnPressBookmark={mockCallback}
      />,
    );
    component.root.findByType(Button).props.onPress();
    expect(mockCallback.mock.calls[0][0]).toBe(id);
  });
});

describe('Maker Popover Component > Android', () => {
  beforeEach(() => {
    jest.mock('Platform', () => {
      const Platform = require.requireActual('Platform');
      Platform.OS = 'android';
      return Platform;
    });
  });
  it('> render the component', () => {
    const component = renderer.create(
      <MakerPopover id="1" name="test" distance={4} rating={3} />,
    );

    expect(component).toMatchSnapshot();
  });
});
