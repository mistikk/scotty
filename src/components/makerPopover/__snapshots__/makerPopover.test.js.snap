// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Maker Popover Component > Android > render the component 1`] = `
<View>
  <Text>
    test
  </Text>
  <Text>
    Distance : 4
  </Text>
  <Text>
    Rating : 3
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text>
      Add to bookmarks
    </Text>
  </View>
</View>
`;

exports[`Maker Popover Component > iOS > render the component 1`] = `
<View>
  <Text>
    test
  </Text>
  <Text>
    Distance : 4
  </Text>
  <Text>
    Rating : 3
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text>
      Add to bookmarks
    </Text>
  </View>
</View>
`;
